wrangler-proto/src/main/java/io/cdap/wrangler/proto/recipe/v2/RecipeId.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.wrangler.proto.recipe.v2;
+
+import io.cdap.cdap.api.NamespaceSummary;
+
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * Recipe id
+ */
+public class RecipeId {
+  private final NamespaceSummary namespace;
```

Comment:
why do we need to store the entire `namespace` summary? `NamespaceSummary` has a description field, which might be large. Also, cdap does store `namespace` summary. Storing this information would be duplication of data. I believe storing namespace as a String would be sufficient.